programs/stake/src/stake_state.rs

```diff
@@ -588,82 +598,54 @@ impl<'a> StakeAccount for KeyedAccount<'a> {
         }
 
         if let StakeState::Uninitialized = split.state()? {
-            // verify enough account lamports
-            if lamports > self.lamports()? {
-                return Err(InstructionError::InsufficientFunds);
-            }
+            let validated_split_info = validate_split_amount(self, split, lamports)?;
 
             match self.state()? {
                 StakeState::Stake(meta, mut stake) => {
                     meta.authorized.check(signers, StakeAuthorize::Staker)?;
-                    let split_rent_exempt_reserve = calculate_split_rent_exempt_reserve(
-                        meta.rent_exempt_reserve,
-                        self.data_len()? as u64,
-                        split.data_len()? as u64,
-                    );
+                    let validated_info = validated_split_info.unwrap();
```

Comment:
Yeah, I was going back and forth on this. Since the impl of `validate_split_info()` guarantees that `unwrap()` is safe here (always `Some` in this variant), I was treating this as a programmer bug and not an instruction error. Maybe an `assert!()` is better here (which is basically what the `unwrap()` is), or a comment? Wdyt?

Reply:
I don't feel that strongly about this, so we can leave it as an `unwrap()` if you think that's better